factory/gen.go

```diff
@@ -133,7 +142,7 @@ func (t *TLS) AddCN(secret *v1.Secret, cn ...string) (*v1.Secret, bool, error) {
 	}
 	secret.Data[v1.TLSCertKey] = certBytes
 	secret.Data[v1.TLSPrivateKeyKey] = keyBytes
-	secret.Annotations[hashKey] = hash
+	secret.Annotations[hashKey] = fmt.Sprintf("SHA1=%X", sha1.Sum(newCert.Raw))
```

Comment:
Why have we down-graded the hash algo?

Reply:
Previously this was a SHA256 hash of not the certificate itself, but the SubjectAlternativeNames attribute text. Since we need to detect and handle changes to the cert as a whole in order to invalidate the caches, we are now hashing all of it which makes this a fingerprint. SHA1 is the standard fingerprint alg for other tools such as `openssl x509`, the Windows certificate management UI, and OS X's Keychain, so using the same alg here makes it easy to compare the certificate information across tools.

Reply:
Would it make sense to have a `const fingerprint = "listener.cattle.io/hash"` and use this as the annotation key instead of re-using the old key which had different semantics?

Reply:
I am 100% down with changing the annotation key. I checked k3s and rancher and it didn't appear that anything else was relying on the annotation, so changing it here should be fine.